### PR TITLE
chore: replace jest with vite

### PR DESCRIPTION
**Pull request by Uzlopak**
replace jest with vite. 

Also ensure we have really 100% code coverage

### CODE REVIEW

File: vite.config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { defineConfig } from 'vite'
+
+export default defineConfig({
```
Comment (wolfy1339) [inline vite.config.js:3]: Can you please lint this file with prettier

Comment (Uzlopak) [reply vite.config.js:3]: @wolfy1339 
done

File: scripts/build.mjs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* c8 ignore start */
```
Comment (wolfy1339) [inline scripts/build.mjs:1]: Can vite not be configured to ignore other directories than `src`?

Comment (Uzlopak) [reply scripts/build.mjs:1]: Ah, sry should be an artifact

**Review by gr2m — APPROVED** [review]
yes, please 🙏🏼💐

**Comment by Uzlopak** [comment]
@gr2m 

Resolved merge conflict

**Comment by wolfy1339** [comment]
This will need to be replicated across the repos

**Pull request merged**